File: app/src/main/kotlin/com/freehands/assistant/data/local/AppDatabase.java

package com.freehands.assistant.data.local;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

@Database(entities = {SampleItem.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract SampleItemDao sampleItemDao();

    private static volatile AppDatabase INSTANCE;
    public static AppDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                        AppDatabase.class, "freehands_db")
                        .fallbackToDestructiveMigration()
                        .build();
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/garbe/freehands/app/src/main/java/com/garbe/freehands/data/local/db/AppDatabase.kt
package com/garbe/freehands.data.local.db

import androidx.room.Database
import androidx.room.RoomDatabase
import com/garbe/freehands.data.local.entity.UserEntity
import com/garbe/freehands.data.local.dao.UserDao

@Database(entities = [UserEntity::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun userDao(): UserDao
}
